### PR TITLE
Feature prepare for release

### PR DESCRIPTION
**Pull request by ritwik-deshpande**
Add .gitignore

Downgraded globus-compute-endpoint to 2.2.0.
We need to use 2.2.0 in-order to run functions using singularity containers.

Updated example. Created venv installed custom-image-builder directly from the feature branch.
Tested the code end to end.

Unified endpoint, single endpoint for building image and running the custom image.

Documented steps for creation of endpoint in delta and using conda env to manage dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+* HPCs do not run docker containers(due to security reasons as discussed [here](https://docs.vscentrum.be/software/singularity.html)) and support only an apptainer/singularity image.
+
+* Installing the apptainer setup to build the singularity image locally is not a straightforward process especially on windows and mac systems as discussed in the [documentation](https://apptainer.org/docs/admin/main/installation.html).
+
+Using this python library the user can specify their custom image specification to build an apptainer/singularity image 
+which would be used to in-turn to run their functions on globus-compute. The library registers the container and 
+returns the container id which would be used by the globus-compute executor to execute the user function.
+
+
+## Prerequisite.
```
Comment (BenGalewsky) [inline README.md:15]: Put the note about GC Endpoint 2.0 here - to make sure users don't ignore your pin in this example.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+1. Create a conda virtual env. We have created a ```custom-image-builder``` conda env on the delta cluster as follows:
+```shell
+conda create --name custom-image-builder python=3.11
+
+conda activate custom-image-builder
+
+pip install globus-compute-endpoint==2.2.0
+
+pip install custom-image-builder
```
Comment (BenGalewsky) [inline README.md]: Tell me we don't need this!

File: README.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        'scheduler_options': '#SBATCH --account=bbmi-delta-cpu',
+    }
+}
+
+config = Config(
+    executors=[
+        HighThroughputExecutor(
+            max_workers_per_node=10,
+            address=address_by_interface('hsn0'),
+            scheduler_mode='soft',
```
Comment (BenGalewsky) [inline README.md:61]: Highlight these three lines in the paragraph. This is the magic that makes it work!

File: README.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    container_id = build_and_register_container(gcc_client=gcc_client,
+                                                endpoint_id=image_builder_endpoint,
+                                                image_file_name="my-pandas-image",
+                                                base_image_type="docker",
+                                                base_image="python:3.8",
+                                                pip_packages=["pandas"])
+
+    print("The container id is", container_id)
+
+    example_endpoint = "0b4e042b-edd5-4951-9ce5-6608c2ef6cb8"
```
Comment (BenGalewsky) [inline README.md]: This should just be the `image_builder_endpoint` - only one endpoint to rule them all!

**Comment by BenGalewsky** [comment]
Remove the polling in the build_container code.
>         while not fut.done():
>            sleep(30)


`fut.result()` should block on the future completing...

**Review by BenGalewsky — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (ritwik-deshpande) [reply README.md]: Yes I'll update the README.md

File: README.md (hunk 2 above)
Comment (ritwik-deshpande) [reply README.md]: Yes I have fixed the issue using from globus_compute_sdk.serialize import DillCodeSource

File: README.md (hunk 1 above)
Comment (ritwik-deshpande) [reply README.md:15]: Yes added it here

**Review by BenGalewsky — APPROVED** [review]
This is good - let's do one final PR with error handling

**Pull request merged**